### ISSUE TITLE
No Window Decorations on Gnome Wayland

### ISSUE THREAD

**Issue by Borges-Filipe** [microsoft/vscode#124202]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.56
- OS Version: Manjaro Gnome (Using Wayland)

Steps to Reproduce:

1. Open VS Code using:

`code --enable-features=UseOzonePlatform --ozone-platform=wayland `

2. Everything looks crisp but there are no window decorations, thus there is no way to move the window using the mouse cursor

**Comment by avanishsubbiah** [comment]
I'm having this issue as well. The window still resizeable with keyboard shortcuts, but without the decorations it's impossible to manage the window with a mouse.

**Comment by major-mayer** [comment]
This is a well known issue. Gnome under Wayland does not support server side decorations, which is a problem for all Electron apps that don't implement client side decorations (which VSCode does).
 
You can enable client side decorations as described here: 
https://github.com/microsoft/vscode/issues/109176#issuecomment-833744341

**Comment by major-mayer** [comment]
And as a little "life-hack" when dealing with those missing window decorations:
You can always move windows by pressing your Super key (mostly the windows one) + left mouse button.

**Comment by major-mayer** [comment]
Yes that's a problem in general with CSD, but unfortunately there won't change much on the Gnome side in the near future...

**Comment by avanishsubbiah** [comment]
> Yes that's a problem in general with CSD, but unfortunately there won't change much on the Gnome side in the near future...

Wait so how do other gtk Wayland applications support the default gnome window title bar?

**Comment by major-mayer** [comment]
Well, they are GTK applications... The GTK toolkit includes methods to draw title bars, decorations, however you call them.

If the developers from Microsoft would really like to add GTK styled window decorations for VSCode running under Gnome and Wayland (which I doubt), it would make more sense to do this in directly inside the Electron project. 
But GTK is a big dependency and I'm not sure if it really makes sense to add a whole toolkit for such a small userbase and only the window decorations(and to satisfy stubborn Gnome developers ;).

AFAIK that's also what prevented the Gnome devs from introducing SSD to the Mutter compositor in the first place. 
They had no dependency on GTK yet, and they won't add them in the future, at least that's what they tell you to justify their point of view... Yeah I don't know, KDE also managed to get it done somehow.
There is a really salty thread on their [Gitlab](https://gitlab.gnome.org/GNOME/mutter/-/issues/217)  where you can read all and long about this, have fun :laughing: 

IMO the only real chance to get out of this stuck situation is something like [libdecoration](https://gitlab.gnome.org/jadahl/libdecoration), a library only for creating window decorations with no dependencies on GTK, QT or whatever.
But I don't know when this finally becomes stable enough for productive use.

**Comment by modzilla99** [comment]
To work around that issue:
https://github.com/microsoft/vscode/issues/109176#issuecomment-846656606
